Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp

```diff
@@ -654,22 +654,7 @@ void GPUConnectionToWebProcess::createRemoteGPU(WebGPUIdentifier identifier, Ren
     auto* renderingBackend = it->value.get();
 
     auto addResult = m_remoteGPUMap.ensure(identifier, [&] {
-        return IPC::ScopedActiveMessageReceiveQueue { RemoteGPU::create(
-#if ENABLE(VIDEO) && PLATFORM(COCOA)
-            [&] (std::variant<WebCore::MediaPlayerIdentifier, WebKit::RemoteVideoFrameReference> identifier, Function<void(RefPtr<WebCore::VideoFrame>)>&& callback) mutable {
-                return WTF::switchOn(identifier, [&] (WebCore::MediaPlayerIdentifier i) {
-                    performWithMediaPlayerOnMainThread(i, [&callback](auto& player) {
-                        callback(player.videoFrameForCurrentTime());
-                    });
-                }, [&] (WebKit::RemoteVideoFrameReference i) {
```

Comment:
So this function is quite incorrect for various reasons.

There's two use-cases that are distinct:
1. Getting content from "MediaPlayer"
2. Getting contetnt from "WebCodecsVideoFrame"

This function conflates the two.

Keep your work distinct, and only concentrate on 2.

For your current patch to improve "the various reasons":
The idea is:
1. WebCodecsVideoFrame video frames are stored in video frame object heap. VideoFrameObjectHeap is thread-safe, it is intended for cross work queue sharing. 
2. WebGPU must fetch the video frame from the object heap, based on the video frame read reference.


To be done:
1. Remove the variant thingy from this
2. Pass Ref<VideoFrameObjectHeap> to WebGPU objects 
3. Read from the video frame object heap whenever you have a video frame read reference.

Communicate the video frame read references from the Web process, do not do it like this thing here, where the read reference is constructed on-the-fly in GPUP based on the identifier.

Reply:
@kkinnunen-apple doesn’t SharedVideoFrameReader already use the RemoteVideoFrameObjecyHeap? I pass it to SharedVideoFrameReader‘s constructor in RemoteDevice.cpp with this patch